src/Dumplie/Application/Query/Customer/Result/Cart.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Application\Query\Customer\Result;
+
+use Dumplie\Application\Exception\InvalidArgumentException;
+
+final class Cart
+{
+    /**
+     * @var CartItem[]|array
+     */
+    private $items;
+
+    /**
+     * @var float
+     */
+    private $totalPrice;
+
+    /**
+     * @param array|CartItem[] $items
+     * @throws InvalidArgumentException
+     */
+    public function __construct(array $items = [])
+    {
+        $this->totalPrice = 0;
+
+        foreach ($items as $item) {
+            if (!$item instanceof CartItem) {
+                throw new InvalidArgumentException();
+            }
+
+            $this->totalPrice += $item->price();
+        }
+
+        $this->items = $items;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isEmpty() : bool
+    {
+        return !count($this->items);
+    }
+
+    /**
+     * @return float
+     */
+    public function totalPrice() : float
+    {
+        return $this->totalPrice;
+    }
+
+    /**
+     * @return string
+     */
+    public function currency() : string
+    {
+        return $this->isEmpty() ? null : $this->items[0]->currency();
```

Comment:
Moim zdaniem należy tutaj założyć odwrotny sposób obsługi produktów i jeżeli sklep obsługuje wiele walut to:

- koszyk trzyma currency sesyjne (lub inne, ale swoje własne)
- produkty dodawane są do koszyka w swoich walutach, każdy z produktów może mieć inną walutę sprzedaży. 
- metody zliczające total muszą operować na currency koszyka i w oparciu o nie wyliczać sumy poszczególnych produktów z uwzględnieniem np. kursu walutowego.